scripts/o3de/o3de/repo.py

```diff
@@ -295,38 +317,77 @@ def search_repo(manifest_json_data: dict,
                 gem_name: str = None,
                 template_name: str = None,
                 restricted_name: str = None) -> dict or None:
-    if isinstance(engine_name, str) or isinstance(engine_name, pathlib.PurePath):
-        o3de_object_uris = manifest_json_data.get('engines', [])
-        manifest_json = 'engine.json'
-        json_key = 'engine_name'
-        search_func = lambda manifest_json_data: manifest_json_data if manifest_json_data.get(json_key, '') == engine_name else None
-    elif isinstance(project_name, str) or isinstance(project_name, pathlib.PurePath):
-        o3de_object_uris = manifest_json_data.get('projects', [])
-        manifest_json = 'project.json'
-        json_key = 'project_name'
-        search_func = lambda manifest_json_data: manifest_json_data if manifest_json_data.get(json_key, '') == project_name else None
-    elif isinstance(gem_name, str) or isinstance(gem_name, pathlib.PurePath):
-        o3de_object_uris = manifest_json_data.get('gems', [])
-        manifest_json = 'gem.json'
-        json_key = 'gem_name'
-        search_func = lambda manifest_json_data: manifest_json_data if manifest_json_data.get(json_key, '') == gem_name else None
-    elif isinstance(template_name, str) or isinstance(template_name, pathlib.PurePath):
-        o3de_object_uris = manifest_json_data.get('templates', [])
-        manifest_json = 'template.json'
-        json_key = 'template_name'
-        search_func = lambda manifest_json_data: manifest_json_data if manifest_json_data.get(json_key, '') == template_name else None
-    elif isinstance(restricted_name, str) or isinstance(restricted_name, pathlib.PurePath):
-        o3de_object_uris = manifest_json_data.get('restricted', [])
-        manifest_json = 'restricted.json'
-        json_key = 'restricted_name'
-        search_func = lambda manifest_json_data: manifest_json_data if manifest_json_data.get(json_key, '') == restricted_name else None
-    else:
-        return None
-    o3de_object = search_o3de_object(manifest_json, o3de_object_uris, search_func)
-    if o3de_object:
-        o3de_object['repo_name'] = manifest_json_data['repo_name']
-        return o3de_object
-
+    repo_schema_version = get_repo_schema_version(manifest_json_data)
+
+    target_name = ''
+
+    if repo_schema_version == "1.0.0":        
```

Comment:
This whole function confuses me a bit, I wonder if there is a way we could simplify it.
Do you know why we are checking if the passed in parameters are strings OR paths? Also it looks like we aren't checking if the parameters are empty or not, perhaps we want to return an empty array/dict if we are looking for all 'engines' in a repo?

I think I can see why the function was made with so many temporary variables and a search function, so it could recurse into multiple repos, but still I wonder if you could replace all the temporaries with a single call to
```
if engine_name:
    o3de_object = search_o3de_object(manifest_json_data, 'engines',' engine.json', 'engine_name')
elif ...

if not o3de_object:
   for repo in manifest_json_data.get('repos',[]):
       o3de_object = search_repo(...)
       if o3de_object:
           break

return o3de_object
```

Reply:
The names are never paths.
They are always strings, so no need for the pathlib.PurePath check.